packages/@aws-cdk/aws-ecs/lib/base/base-service.ts

```diff
@@ -23,6 +24,37 @@ export interface IService extends IResource {
   readonly serviceArn: string;
 }
 
+export interface EcsTarget {
+  /**
+   * The name of the container.
+   */
+  readonly containerName: string;
+
+  /**
+   * The port number of the container. Only applicable when using application/network load balancers.
+   *
+   * @default - Container port of the first added port mapping.
+   */
+  readonly containerPort?: number;
+
+  /**
+   * The protocol used for the port mapping. Only applicable when using application load balancers.
+   *
+   * @default Protocol.TCP
+   */
+  readonly protocol?: Protocol;
+
+  /**
+   * ID for a target group to be created.
+   */
+  readonly newTargetGroupId: string;
```

Comment:
Can we rename this to targetGroupId (new is relative)

Reply:
Ha, I just made him change it from `targetGroupId` -> `newTargetGroupId`.

The thing is, I'm not sure that `targetGroupId` conveys accurately enough that it will be created, rather than one that already exists that happens to have that ID.

Reply:
Fair enough, just find newTargetGroup to be ambiguous.